heaps/min_heap.py

```diff
@@ -57,18 +70,42 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: O(log n)
+            Space complexity: O(log n)
         """
-        pass
+        if index == 0:
+            return
+        if self.store[index].key < self.store[(index - 1) // 2].key:
+            self.swap(index, (index - 1) // 2)
+            self.heap_up((index - 1) // 2)
 
     def heap_down(self, index):
         """ This helper method takes an index and 
             moves the corresponding element down the heap if it's 
             larger than either of its children and continues until
             the heap property is reestablished.
         """
-        pass
+
+        if index * 2 + 1 < len(self.store):
+            if index * 2 + 2 < len(self.store):
+                if self.store[index * 2 + 1].key > self.store[index * 2 + 2].key:
+                    if self.store[index * 2 + 1].key < self.store[index].key:
```

Comment:
👀 The one we want to swap is the smaller of the two children. This will swap the larger of the two children. This is the other part causing the test failure for the remove ordering.